Dockerfile

```diff
@@ -0,0 +1,21 @@
+FROM golang:1.23.3-alpine3.20
+
+LABEL mantainer="antoniorr@correo.ugr.es" version="5.0.0"
+
+WORKDIR /app
+
+RUN adduser -D -u 1001 test && chown test /app
```

Comment:
¿Por qué necesitas asignarle el uid 1001?

Reply:
Esto lo añadí porque el profesor nos comentó durante la explicación del objetivo que era necesario utilizar este usuario.

Reply:
Por favor, dejad de decir "esto lo comentó el profesor". No hay ni una sola ocasión en la que justifiquéis algo porque "lo haya dicho el profesor" hayáis entendido correctamente lo que se ha dicho ni por qué (tampoco preguntáis el por qué).
También sabéis que cuando se hace un comentario en un PR, hay que reflexionar y cambiar algo. Una respuesta a la pregunta *nunca* justifica el resolver la conversación.

Reply:
Eso hablamos en clase que es por seguridad, ya que no es una buena práctica usar el usuario root que tiene todos los permisos, con ello, para poder ejecutar con el usuario test, es recomendable asignarle al mismo este UID.  ¿Debería hacer con ello algo al respecto @JJ ?

Reply:
Lo que debes hacer es tratar de pensar en una respuesta al comentario de @hossam1522  más allá de "me lo dijo el profesor". Os dije *el primer día de clase* que os teníais que preguntar el por qué de las cosas. Si esa es la única respuesta que tienes, es que no sabes por qué lo has puesto. Y si estás usando código sin saber por qué razón, es mejor que lo borres o trates de entender la razón por la que se hace.
También sería conveniente que tomárais apuntes cuando os hablo, porque *casi siempre* explico el por qué de las cosas. Y si no lo explico, lo preguntáis.

Reply:
Buenas de nuevo, en primer lugar gracias por las recomendaciones, lo de tomar apuntes lo hice y lo que copié de forma rápida y esquemática relativo a este apartado es lo que sigue:

"- USUARIO no privilegiado --> Si escogemos una imagen con un SO generalmente vamos a tener un usuario predefinido con su home en el que vamos a tener permisos de escritura. ---> Se debe de ejecutar con el uid 1001"

De hecho tanto es así, que si no lo incluyo el resultado es el siguiente:
┌──(antonio㉿kali)-[~/…/quinto/informatica/IV/MenuConsulter]
└─$ docker run -u 1001 -t -v `pwd`:/app/test imagen
echo "Ejecutando tests del proyecto..."
Ejecutando tests del proyecto...
go test ./internal -v
failed to initialize build cache at /.cache/go-build: mkdir /.cache: permission denied
error: Recipe `test` failed on line 29 with exit code 1

Tal y como se ve falla al no tener permisos de escritura el usuario test  donde Go intenta crear su caché de compilación

En cualquier caso he estado buscando un buen rato para ver si es posible no utilizar este uid y he obtenido otra posible solución, creando la carpeta donde el proyecto guardará su caché de compilación mediante una variable de entorno y sobre la cuál el usuario tiene permisos.

Con todo ello, resuelto en: [82f6902](https://github.com/antoniorr02/MenuConsulter/pull/28/commits/82f6902f18f1acda181a12cff0e8f39058c2b172)

Reply:
> Buenas de nuevo, en primer lugar gracias por las recomendaciones, lo de tomar apuntes lo hice y lo que copié de forma rápida y esquemática relativo a este apartado es lo que sigue:
> 
> "- USUARIO no privilegiado --> Si escogemos una imagen con un SO generalmente vamos a tener un usuario predefinido con su home en el que vamos a tener permisos de escritura. ---> Se debe de ejecutar con el uid 1001"
> 

Efectivamente. ¿Qué quiere decir *se debe de ejecutar* entonces?

> De hecho tanto es así, que si no lo incluyo el resultado es el siguiente: ┌──(antonio㉿kali)-[~/…/quinto/informatica/IV/MenuConsulter] └─$ docker run -u 1001 -t -v `pwd`:/app/test imagen echo "Ejecutando tests del proyecto..." Ejecutando tests del proyecto... go test ./internal -v failed to initialize build cache at /.cache/go-build: mkdir /.cache: permission denied error: Recipe `test` failed on line 29 with exit code 1

Correcto.